docs/appnotes/0016-authorisation-in-tams-workflows.md

```diff
@@ -0,0 +1,308 @@
+| Source News X  | `news`, `sport_ro` | News have full access. Sport have read access only. |
+| Source News Y  | `news`             | News have full access. Sport have no access.        |
+
+### Auth logic
+
+In order that implementations may have consistent expectations about which methods they may access, this section provides recommended auth logic for methods.
+
+It is assumed that admins have permission to execute all methods on all endpoints.
+It is only explicitly called out in the listing below where admins are the only users granted permissions.
+
+The listing below refers to requests having permissions, rather than users.
+This is to account for cases where users only "claim" a subset of their permissions for a given request.
+Note that in some circumstances, requests may have to claim more permissions than may initially be assumed.
+For example - when editing the `auth_classes` tag on a Source/Flow/webhook, requests must claim both write permissions and the permission they are changing.
+i.e. If the request adds or removes delete permissions for any group, it must have valid delete permissions itself.
+This is to prevent permission escalation attacks such as a user with write permissions adding delete permissions to themselves.
+
+Implementations may choose to additionally filter data based on the permissions of a request.
+For example, where a Source collections may be filtered to only include Sources the request has read/write/delete permissions on.
+Implementers should consider the implications of hiding data.
+For example - hiding collection relationships may result in clients deciding to delete a resource which, unknowingly, is still referenced by another.
+
+| Endpoint                             | Method       | Auth logic                                                                              |
+| ------------------------------------ | ------------ | --------------------------------------------------------------------------------------- |
+| `/`                                  | `HEAD`/`GET` | Available to all                                                                        |
+| `/service`                           | `HEAD`/`GET` | Available to all                                                                        |
+|                                      | `POST`       | Request must have admin permissions. Otherwise reject.                                  |
+| `/service/storage-backends`          | `HEAD`/`GET` | Available to all                                                                        |
+| `/service/webhooks`                  | `HEAD`/`GET` | Restrict returned data by adding list of claimed auth classes to `tag.auth_classes.includes`. If the incoming request has `tag.auth_classes.includes` set, the request must be processed with `tag.auth_classes.includes` set to the intersection of the claimed auth classes and the provided list in `tag.auth_classes.includes`. |
```

Comment:
The implementations has "includes" as the default behaviour - does the `.includes` need to be removed here (and in other similar rows)